samples/ControlCatalog/App.xaml

```diff
@@ -1,6 +1,17 @@
 <Application xmlns="https://github.com/avaloniaui"
              xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
              x:Class="ControlCatalog.App">
+  <TrayIcon.TrayIcons>
+    <TrayIcons>
+      <TrayIcon Icon="/Assets/test_icon.ico" ToolTipText="Avalonia Tray Icon ToolTip">
+        <NativeMenu.Menu>
+          <NativeMenu>
+            <NativeMenuItem Header="About" />
+          </NativeMenu>
+        </NativeMenu.Menu>
+      </TrayIcon>
+    </TrayIcons>
```

Comment:
With Flyout API it should be possible to do in this way:

```suggestion
    <TrayIcons>
      <TrayIcon Icon="/Assets/test_icon.ico" ToolTipText="Tray Icon with button">
        <Flyout>
           <Button />
        </Flyout>
      </TrayIcon>
      <TrayIcon Icon="/Assets/test_icon.ico" ToolTipText="Tray Icon with menu">
        <MenuFlyout>
           <MenuItem Header="About" />
        </MenuFlyout>
      </TrayIcon>
      <TrayIcon Icon="/Assets/test_icon.ico" ToolTipText="Tray Icon with native menu (will be mapped into NativeMenu)">
        <MenuFlyout UseNativeMenu="True">
           <MenuItem Header="About" />
        </MenuFlyout>
      </TrayIcon>
    </TrayIcons>
```

Reply:
Although using NativeMenu could be less painful in short term

Reply:
We can only use nativemenu, because Linux only supports menus, so we can only support Native tray menus.

Reply:
@danwalmsley a curiosity. In Linux it is not possible to intercept the click on the treyicon and open a popup window?

Reply:
yes. it seems osx is the same... probably going to remove those apis.

Reply:
Thanks for the reply. Have you also considered the possibility of having both NativeMenu and Menu?

Reply:
Menu supports many more scenarios than NativeMenu, and its also a control and needs to be connected to the tree.

Where as NativeMenu is simply a model, it can exist outside the tree.

NativeMenus can be exported / converted to Menus though.

Reply:
thanks for explaining